### PR TITLE
Add market parameter to Perpetual

### PR DESCRIPTION
**Pull request by Kenadia**

### CODE REVIEW

File: scripts/SaveDeployedAddresses.ts [hunk 1]
```diff
@@ -1,13 +1,11 @@
-import { promisify } from 'es6-promisify';
```
Comment (BrendanChou) [inline scripts/SaveDeployedAddresses.ts]: does anything use promisify now? or can we remove it from `package.json`?

File: scripts/SaveDeployedAddresses.ts [hunk 2]
```diff
@@ -17,29 +15,32 @@ if (!Object.values(PerpetualMarket).includes(MARKET_PAIR)) {
...
       }
     });
   });
 
-  const json = JSON.stringify(deployed, null, 4) + '\n';
+  const json = `${JSON.stringify(deployed, null, 4)}\n`;
 
   const filepath = `${__dirname}/../migrations/deployed.json`;
-  await writeFileAsync(filepath, json);
+  await fs.writeFileSync(filepath, json);
```
Comment (BrendanChou) [inline scripts/SaveDeployedAddresses.ts:45]: nit: await not needed?

File: scripts/CleanBuild.ts [hunk 3]
```diff
@@ -25,36 +31,43 @@ async function clean(): Promise<void> {
...
+        );
+      }
+    });
 
-    const filename = `${contractName}.json`;
-    await writeFileAsync(directory + filename, json);
+    // Write cleaned JSON artifacts to build/published_contracts.
+    const publishJson = JSON.stringify(cleaned, null, 4);
+    const publishFilepath = `${publishDir}${contractName}.json`;
+    await fs.writeFileSync(publishFilepath, publishJson);
```
Comment (BrendanChou) [inline scripts/CleanBuild.ts:58]: nit: await not needed?

File: scripts/CleanBuild.ts [hunk 4]
```diff
@@ -25,36 +31,43 @@ async function clean(): Promise<void> {
...
-
-  await Promise.all(promises);
+    // Copy deployed contract information, with market pairs, to build/contracts.
+    const buildFilepath = `${buildDir}${contractName}.json`;
+    contract.networks = {
+      ...contract.networks,
+      ...cleaned.networks,
+    };
+    const buildJson = JSON.stringify(contract, null, 4);
+    await fs.writeFileSync(buildFilepath, buildJson);
```
Comment (BrendanChou) [inline scripts/CleanBuild.ts:68]: nit: await not needed for sync?

File: scripts/SaveDeployedAddresses.ts (hunk 1 above)
Comment (Kenadia) [reply scripts/SaveDeployedAddresses.ts]: It's used in SignatureHelper, but I think we can use `util.promisify` instead. Just updated.

**Pull request merged**